src/include/nodes/parsenodes.h

```diff
@@ -327,6 +327,7 @@ typedef struct TypeCast
 	NodeTag		type;
 	Node	   *arg;			/* the expression being casted */
 	TypeName   *typeName;		/* the target type */
+	bool		is_trycast;		/* TODO */
```

Comment:
This ABI change may lead to numerous failures, both instant(when gpstart) and deferred (executing query with view). 
You need to reuse existing fields within TypeCast struct. 
One option is to prepend typeName with `__opengpdb_trycast__` prefix